assayist/processor/base.py

```diff
@@ -0,0 +1,184 @@
+
+
+class Analyzer(ABC):
+    """Base Abstract class that analyzers will inherit from."""
+
+    METADATA_DIR = '/metadata/'
+    MESSAGE_FILE = METADATA_DIR + 'message.json'
+    BUILD_FILE = METADATA_DIR + 'buildinfo.json'
+    TASK_FILE = METADATA_DIR + 'taskinfo.json'
+    RPM_FILE = METADATA_DIR + 'rpms.json'
+    ARCHIVE_FILE = METADATA_DIR + 'archives.json'
+    IMAGE_RPM_FILE = METADATA_DIR + 'image-rpms.json'
+    BUILDROOT_FILE = METADATA_DIR + 'buildroot-components.json'
+
+    def main(self):
+        """Call this to run the analyzer."""
+        neomodel.db.set_connection(config.DATABASE_URL)
+        # run the analyzer in a transaction
+        neomodel.db.begin()
+        try:
+            self.run()
+            log.debug('Analyzer completed successfully, committing.')
+            neomodel.db.commit()
+        except Exception as e:
+            log.exception('Error encountered executing Analyzer, rolling back transaction.')
+            neomodel.db.rollback()
+            raise
+
+    @abstractmethod
+    def run(self):
+        """Implement analyzer code here in your subclass."""
+
+    def read_metadata_file(self, in_file):
+        """
+        Read and return the specified json metadata file or an empty dict.
+
+        :param str in_file: The path to the input file to read. Probably one of the class constants.
+        :return: a dict or list read from the file, or an empty dict
+        :rtype: {}
+        :raises ValueError: if the file was not valid json content
+        """
+        if os.path.isfile(in_file):
+            with open(in_file, 'r') as f:
+                return json.load(f)
+        else:
+            log.debug('File not found: %s, returning empty dict', in_file)
+            return {}
+
+    def get_or_create_build(self, build_id, build_type):
+        """
+        Get or create a build object.
+
+        :param str build_id: the id of the build
+        :param str build_type: the type of the build (eg. "build", "maven", "buildContainer")
+        :return: a Build object
+        :rtype: assayist.common.models.content.Build
+        """
+        return content.Build.get_or_create({
+            'id_': build_id,
+            'type_': build_type})[0]
+
+    def get_or_create_rpm_artifact(self, id, name, version, release, arch, checksum):
+        """
+        Create or update an Artifact for this rpm.
+
+        :param str id: the rpm's id
+        :param str name: the rpm's name, eg 'kernel'
+        :param str version: the rpm's version, eg '1'
+        :param str release: the rpm's release, eg '2.el7'
+        :param str arch: the rpm's arch, eg 'x86_64'
+        :param str checksum: the rpm's checksum, eg 'deadbeef1234'
+        :return: an Artifact object
+        :rtype: assayist.common.models.content.Artifact
+        """
+        filename = f'{name}-{version}-{release}.{arch}.rpm'
+        if arch == 'src':
+            type = content.Artifact.TYPES['srpm']
```

Comment:
The value of `content.Artifact.TYPES['srpm']` is the display value, therefore, the key is what should be stored as the type.

Reference:
https://neomodel.readthedocs.io/en/latest/properties.html#choices